realm/realm-library/src/androidTest/java/io/realm/RealmTests.java

```diff
@@ -4792,4 +4793,38 @@ public void getNumberOfActiveVersions() throws InterruptedException {
         } catch (IllegalStateException ignore) {
         }
     }
+
+    @Test
+    public void getCachedInstanceDoNotTriggerStrictMode() {
+        StrictMode.setThreadPolicy(new StrictMode.ThreadPolicy.Builder()
+            .detectAll()
+            .penaltyLog()
+            .penaltyDeath()
+            .build());
+        try {
+            Realm.getInstance(realmConfig).close();
+        } finally {
+            StrictMode.setThreadPolicy(new StrictMode.ThreadPolicy.Builder()
+                    .permitAll()
+                    .build());
+        }
+    }
+
+    @Test
+    public void getCachedInstanceFromOtherThreadDoNotTriggerStrictMode() throws InterruptedException {
+        StrictMode.setThreadPolicy(new StrictMode.ThreadPolicy.Builder()
+                .detectAll()
+                .penaltyLog()
+                .penaltyDeath()
+                .build());
+        try {
+            Thread t = new Thread(() -> Realm.getInstance(realmConfig).close());
```

Comment:
should you capture potential exception thrown in thread `t`?

Reply:
There shouldn't be a need. An exception in this thread will correctly fail the test.